packages/headless/src/controllers/product-listing/range-facet/date-facet/headless-produc-listing-date-facet.test.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  DateFacet,
+  buildDateFacet,
+  DateFacetOptions,
+} from './headless-product-listing-date-facet';
+import {
+  MockProductListingEngine,
+  buildMockProductListingEngine,
+} from '../../../../test/mock-engine';
+import {buildMockDateFacetValue} from '../../../../test/mock-date-facet-value';
+import {buildMockDateFacetRequest} from '../../../../test/mock-date-facet-request';
+import {configuration, dateFacetSet, search} from '../../../../app/reducers';
+import {DateFacetValue} from '../../../../features/facets/range-facets/date-facet-set/interfaces/response';
+import {ProductListingAppState} from '../../../../state/product-listing-app-state';
+import {buildMockProductListingState} from '../../../../test/mock-product-listing-state';
+import {fetchProductListing} from '../../../../features/product-listing/product-listing-actions';
+
+describe('date facet', () => {
+  const facetId = '1';
+  let options: DateFacetOptions;
+  let state: ProductListingAppState;
+  let engine: MockProductListingEngine;
+  let dateFacet: DateFacet;
+
+  function initDateFacet() {
```

Comment:
I don't think its necessary to encapsulate these two in their own function, the rest of the stuff is done directly in the beforeEach